### PR TITLE
feat(suite-native): Grouping assets in account screen

### LINKED ISSUE #7072: List of assets: group assets by coin
Ad a user I would like to see my assets grouped by coin so that I can have better clarity of my assets
Figma: 

Image

Acceptance criteria:
All assets are grouped by coin and separated in the box

Out of scope: 
Search is not part of this issue, please do not even put the icon there

### PR DESCRIPTION
**Pull request by juriczech**
## Description
Refactor `My Assets` page to group accounts from same asset type. Add ability to redirect user to adding another coin straight from this screen so some navigation refactoring was done.

To test - just check design change and try the add button from the header. 

## Related Issue

Resolve [List of assets: group assets by coin#7072](https://github.com/trezor/trezor-suite/issues/7072)

## Screenshots:
image

### CODE REVIEW

File: suite-native/accounts/src/components/AccountsListGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { Box } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { AccountsRootState, selectAccountsByNetworkSymbols } from '@suite-common/wallet-core';
+import { NetworkSymbol } from '@suite-common/wallet-config/libDev/src';
```
Comment (bouchja1) [inline suite-native/accounts/src/components/AccountsListGroup.tsx]: whoopsie `/libDev/src`

**Comment by bouchja1** [comment]
Btw I tried it with "Reset storage" and the app has some troubles with initialization. It seems it is because of the new **origin** navigation param:

Snímek obrazovky 2022-12-08 v 16 54 00

File: suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx [hunk 2]
```diff
@@ -92,7 +92,7 @@ export const XpubScanScreen = ({
     };
 
     return (
-        <Screen header={<AccountImportHeader activeStep={1} />}>
+        <Screen header={<AccountImportHeader origin={route.params.origin} activeStep={1} />}>
```
Comment (bouchja1) [inline suite-native/module-accounts-import/src/screens/XpubScanScreen.tsx]: This will fail when the app is in "onboarding mode" - like a first run of the app.

File: suite-native/module-accounts-import/src/components/AccountImportHeader.tsx [hunk 3]
```diff
@@ -32,18 +34,30 @@ type NavigationProp = StackToTabCompositeProps<
...
 
-    const handleCloseOnboarding = () =>
-        navigation.navigate(RootStackRoutes.AppTabs, {
-            screen: AppTabsRoutes.HomeStack,
-            params: {
-                screen: HomeStackRoutes.Home,
-            },
-        });
+    const handleCloseOnboarding = () => {
+        if (origin === HomeStackRoutes.Home) {
```
Comment (bouchja1) [inline suite-native/module-accounts-import/src/components/AccountImportHeader.tsx]: nit: I personally prefer switch with a default case for more than two conditions because of better readability.

File: suite-native/navigation/src/components/Screen.tsx [hunk 4]
```diff
@@ -4,12 +4,14 @@ import { useSafeAreaInsets, EdgeInsets } from 'react-native-safe-area-context';
...
 import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
 import { Color, nativeSpacings } from '@trezor/theme';
+import { Box, Divider } from '@suite-native/atoms';
 
 import { ScreenContent } from './ScreenContent';
 
 type ScreenProps = {
     children: ReactNode;
     header?: ReactNode;
+    isHeaderDivided?: boolean;
```
Comment (bouchja1) [inline suite-native/navigation/src/components/Screen.tsx]: nit:

```suggestion
    hasDivider?: boolean;
```

**Review by bouchja1 — CHANGES_REQUESTED** [review]
Some nitpicks, but there is a problem with the app on first launch due to `origin`.

File: suite-native/accounts/src/components/AccountsListGroup.tsx (hunk 1 above)
Comment (bouchja1) [reply suite-native/accounts/src/components/AccountsListGroup.tsx]: @juriczech it is still there

File: suite-native/module-accounts-import/src/components/AccountImportHeader.tsx [hunk 5]
```diff
@@ -32,18 +34,33 @@ type NavigationProp = StackToTabCompositeProps<
...
-                screen: HomeStackRoutes.Home,
-            },
-        });
+    const handleCloseOnboarding = () => {
+        switch (origin) {
+            case HomeStackRoutes.Home:
+                navigation.navigate(RootStackRoutes.AppTabs, {
+                    screen: AppTabsRoutes.HomeStack,
+                    params: {
+                        screen: HomeStackRoutes.Home,
```
Comment (bouchja1) [inline suite-native/module-accounts-import/src/components/AccountImportHeader.tsx]: What exactly is this `screen` param? I can't see it in `navigation/src/navigators.ts`

**Review by bouchja1 — CHANGES_REQUESTED** [review]
libDev in import is still there and one question about navigation param

File: suite-native/accounts/src/components/AccountsListGroup.tsx (hunk 1 above)
Comment (juriczech) [reply suite-native/accounts/src/components/AccountsListGroup.tsx]: https://github.com/trezor/trezor-suite/pull/7149/commits/eff4135b3fe9d660157249245a39d34c6c0ebeff

**Review by bouchja1 — APPROVED** [review]

File: suite-native/navigation/src/components/ScreenContent.tsx [hunk 6]
```diff
@@ -16,16 +16,13 @@ const screenContentStyle = prepareNativeStyle<{
     insets: EdgeInsets;
     customHorizontalPadding: number;
     customVerticalPadding: number;
-}>((_, { insets, customHorizontalPadding, customVerticalPadding }) => {
-    const { top, right, bottom, left } = insets;
-    return {
-        flexGrow: 1,
-        paddingTop: Math.max(top, customVerticalPadding),
```
Comment (Nodonisko) [inline suite-native/navigation/src/components/ScreenContent.tsx:23]: Why did you delete this?

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-native/navigation/src/components/ScreenContent.tsx (hunk 6 above)
Comment (juriczech) [reply suite-native/navigation/src/components/ScreenContent.tsx:23]: there was no need to add any insets to the content itself so I removed it as unnecessary logic

Comment (Nodonisko) [reply suite-native/navigation/src/components/ScreenContent.tsx:23]: This is quite important because without it can mess up on devices with bigger insets.

Comment (juriczech) [reply suite-native/navigation/src/components/ScreenContent.tsx:23]: But the insets are only being used in the "corners of the device" right? So if the content wrapper has this implemented than the content itself doesn't need to have this implemented. I deleted it under this assumption

Comment (juriczech) [reply suite-native/navigation/src/components/ScreenContent.tsx:23]: I can return it, it's no big deal. I'm just curious

**Review by Nodonisko — APPROVED** [review]
